heat_diffusion_experiment/python_version.py

```diff
@@ -32,3 +33,86 @@ def python_version(M, num_timesteps=0, total_time=500.0):
             for j in xrange(1,M-1):
                 u[i,j] = (f*u0[i,j+1] + f*u0[i,j-1] + f*u0[i+1,j] + f*u0[i-1,j] + B*u0[i,j]) / (B + 4*f)
     return u
+
+@njit(float64[:, :](int64, int64, float64))
+def numba_version(M, num_timesteps=0, total_time=500.0):
+    # Physical properties --------------------------------------------------------------------------------

```

Comment:
Just to be sure: Are the two implementations exactly the same?

Reply:
That was my intention, for numba version I just plain copied and pasted the python original code. For cython I had to type the variables but the main computation should be equal:

```python
u[i,j] = (f*u0[i,j+1] + f*u0[i,j-1] + f*u0[i+1,j] + f*u0[i-1,j] + B*u0[i,j]) / (B + 4*f)
```